src/test/groovy/com/lesfurets/jenkins/unit/declarative/TestDeclarativePipeline.groovy

```diff
@@ -23,6 +23,12 @@ class TestDeclarativePipeline extends DeclarativePipelineTest {
         assertJobStatusSuccess()
     }
 
+    @Test void jenkinsfile_thisObject() throws Exception {
+        def script = runScript('Declarative_thisObject_JenkinsFile')
+        printCallStack()
+        assertJobStatusSuccess()
```

Comment:
it needs additional asserts to verify exactly the output

Reply:
Currently we use it from jenkins to setup a pipeline script as
functionThatHasPipeline()

This due to how https://www.jenkins.io/doc/book/pipeline/shared-libraries/#defining-custom-steps says custom steps work.
We can even have some params which we can use in the pipeline block this way.

So the attempt was to have something that looks like my own project to make sure this works, as it works within jenkins. Though within jenkins the need for call() at the bottom is not necessary. It seems tests expect a run() method and jenkins uses a call().

Reply:
I added an assert and an echo in the method